### ISSUE TITLE
Panic On Start

### ISSUE THREAD

**Issue by AldaronLau** [ardaku/alloy#36]
```console
Panic happened

--
called `Option::unwrap()` on a `None` value
--

Location src/image_cache/directory.rs:342:43

   0: alloy::handle_panic::handle_panic
   1: std::panicking::rust_panic_with_hook
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:702:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:586:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
   4: rust_begin_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
   5: core::panicking::panic_fmt
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
   6: core::panicking::panic
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:48:5
   7: alloy::image_cache::directory::Directory::check_filter_ready
   8: alloy::playback_manager::PlaybackManager::update_image
   9: <alloy::widgets::picture_widget::PictureWidget as gelatin::Widget>::before_draw
  10: <gelatin::line_layout_container::LineLayoutContainer<Dim> as gelatin::Widget>::before_draw
  11: <gelatin::line_layout_container::LineLayoutContainer<Dim> as gelatin::Widget>::before_draw
  12: gelatin::window::Window::main_events_cleared
  13: gelatin::application::Application::start_event_loop::{{closure}}
  14: winit::platform_impl::platform::wayland::event_loop::EventLoop<T>::run
  15: winit::platform_impl::platform::EventLoop<T>::run
  16: winit::event_loop::EventLoop<T>::run
  17: gelatin::application::Application::start_event_loop
  18: alloy::main
  19: std::sys_common::backtrace::__rust_begin_short_backtrace
  20: std::rt::lang_start::{{closure}}
  21: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:48
      std::panicking::try::do_call
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
      std::panicking::try
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
      std::panic::catch_unwind
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
      std::rt::lang_start_internal
             at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:20
  22: main
  23: __libc_start_main
  24: _start

===================================================================================================
```

**Comment by wdv4758h** [comment]
I tried the current stable branch `97df73fe1f264f04cf6967fa5565d0ed820fed97` on Wayland (Sway 1.8.1). It crashed on start. However, I do find a fork of emulsion that works on my machine.

In case it helps in some ways. The Wayland/Sway working fork I found is [this one](https://github.com/ChristophHess/emulsion).

**Comment by AldaronLau** [comment]
@wdv4758h I think the Wayland crash is unrelated to this crash.  I expect your crash should be fixed when I resolve https://github.com/ardaku/alloy/issues/71.

I'm guessing you have to enable [the `wayland` feature](https://github.com/ChristophHess/emulsion/blob/9485f611e7adb21d99e6f7ff4c2dc662ff76bb55/Cargo.toml#L62)?  Or does it work without?

**Comment by wdv4758h** [comment]
> @wdv4758h I think the Wayland crash is unrelated to this crash. I expect your crash should be fixed when I resolve #71.
> 
> I'm guessing you have to enable [the `wayland` feature](https://github.com/ChristophHess/emulsion/blob/9485f611e7adb21d99e6f7ff4c2dc662ff76bb55/Cargo.toml#L62)? Or does it work without?

Actually, both works, it works with or without `wayland` feature. Not sure what's behavior difference, but that repo works on Sway 1.8.1 out of the box. No crashing and can view images normaly.

**Comment by AldaronLau** [comment]
> > I'm guessing you have to enable [the `wayland` feature](https://github.com/ChristophHess/emulsion/blob/9485f611e7adb21d99e6f7ff4c2dc662ff76bb55/Cargo.toml#L62)? Or does it work without?
> 
> Actually, both works, it works with or without `wayland` feature. Not sure what's behavior difference, but that repo works on Sway 1.8.1 out of the box. No crashing and can view images normaly.

Interesting, I'd have to look closer to see what changed.  Thanks for the info!
